### PR TITLE
[feat]: [CI-4755]: Add support for publishing status check to git via manager

### PR DESCRIPTION
**Pull request by shubham149**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34533)

### CODE REVIEW

**Comment by shubham149** [comment]
trigger messagecheck

**Comment by shubham149** [comment]
trigger ti0

File: 933-scm-commons/src/main/java/io/harness/git/checks/GitStatusCheckParams.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import io.harness.delegate.beans.ci.pod.ConnectorDetails;
+import io.harness.delegate.task.ci.GitSCMType;
+
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class GitStatusCheckParams {
+  private String title;
```
Comment (harshjain12) [inline 933-scm-commons/src/main/java/io/harness/git/checks/GitStatusCheckParams.java:12]: private is not required with @value

**Review by harshjain12 — DISMISSED** [review]
:lgtm:

> Reviewed 7 of 11 files at r2, 7 of 12 files at r3, 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34533#-:-N5FL8Ac5t7j0lJpIFq4:bnfp4nl)* status: :shipit: complete! ✓

**Review by abhinav-harness — DISMISSED** [review]
> Reviewed 7 of 11 files at r2, 6 of 12 files at r3, 5 of 5 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34533)* status: :shipit: complete! ✓

**Comment by jennyjames21** [comment]
*[`260-delegate/BUILD.bazel` line 25 at r4](https://reviewable.io/reviews/harness/harness-core/34533#-N5H_nrJ8G03NRHVRO7r:-N5H_nrJ8G03NRHVRO7s:bh4hkx1) ([raw file](https://github.com/harness/harness-core/blob/1a0138579409f634500faa92a5038bbdcba6ae9e/260-delegate/BUILD.bazel#L25)):*
> ```Starlark
>         "//420-delegate-service:module",
>         "//490-ce-commons:module",
>         "//933-scm-commons:module",
> ```

you already have this dependency on 930, so you dont need it here.  We just cleaned up and move all dependency to 930.  



_Code quote (from 930-delegate-tasks/BUILD.bazel):_
```Starlark
"//933-scm-commons:module",
```

**Review by jennyjames21 — COMMENTED** [review]
> Reviewed 6 of 11 files at r2, 5 of 12 files at r3, 4 of 5 files at r4, 2 of 2 files at r5, 1 of 1 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34533)* status: 0 of 1 approvals obtained.

**Review by shubham149 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34533)* status: 2 file(s) to review. 2. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`260-delegate/BUILD.bazel` line 25 at r4](https://reviewable.io/reviews/harness/harness-core/34533#-N5H_nrJ8G03NRHVRO7r:-N5J8c6n4gIpp5l3swxH:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1a0138579409f634500faa92a5038bbdcba6ae9e/260-delegate/BUILD.bazel#L25)):*
<details><summary><i>Previously, jennyjames21 (Jenny James) wrote…</i></summary><blockquote>

you already have this dependency on 930, so you dont need it here.  We just cleaned up and move all dependency to 930.  
</blockquote></details>

Done.
___
*[`933-scm-commons/src/main/java/io/harness/git/checks/GitStatusCheckParams.java` line 12 at r4](https://reviewable.io/reviews/harness/harness-core/34533#-N5Ey_lsjY20dWnQdgy3-r4-12:-N5J8bP25kjJIqJAQe-y:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/1a0138579409f634500faa92a5038bbdcba6ae9e/933-scm-commons/src/main/java/io/harness/git/checks/GitStatusCheckParams.java#L12)):*
<details><summary><i>Previously, harshjain12 (Harsh Jain) wrote…</i></summary><blockquote>

private is not required with @value
</blockquote></details>

Done.

**Review by shubham149 — COMMENTED** [review]
> Reviewed 6 of 11 files at r2, 4 of 12 files at r3, 3 of 5 files at r4, 2 of 2 files at r5, 1 of 1 files at r6, 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34533)* status: 2 file(s) to review. 2. unresolved discussions. 0 of 1 approvals obtained.

**Comment by shubham149** [comment]
trigger codebasehash

**Review by jennyjames21 — APPROVED** [review]
> Reviewed 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34533)* status: 1. unresolved discussions.

**Comment by shubham149** [comment]
trigger ti2

**Comment by shubham149** [comment]
trigger ti2

**Comment by shubham149** [comment]
trigger ti2

**Review by harshjain12 — APPROVED** [review]
:lgtm:

> Reviewed 1 of 2 files at r5, 1 of 1 files at r6, 2 of 2 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34533#-:-N5KisXE9k-Rvl032utc:bnfp4nl)* status: 1. unresolved discussions.

**Review by harshjain12 — COMMENTED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34533#-:-N5KiuQ69Mb4jflqwLzq:bnfp4nl)* status: 1. unresolved discussions.

**Pull request merged**